libraries/AP_Logger/AP_Logger_MAVLink.cpp

```diff
@@ -550,6 +576,15 @@ void AP_Logger_MAVLink::periodic_1Hz()
         _sending_to_client = false;
         return;
     }
+    // warn user if we can't process the logs quickly enough
+    uint8_t percent_dropped = 0;
+    if (stats.state_dropped > 0) {
+        percent_dropped = (stats.state_dropped * 100) / (stats.state_sent + stats.state_dropped);
+    }
+    if (_dropped_message_counter++ > 5 && percent_dropped > 10) {
+        gcs().send_text(MAV_SEVERITY_WARNING, "MAVLink dataflash logging is dropping blocks (%u%% dropped). Reduce SCHED_LOOP_RATE or LOG_BITMASK.", percent_dropped);
```

Comment:
We already transport dropped-blocks in sys_status where a GCS can plot them

This diagnostic could also be done on the client side (which, being a mavlink node, should be able to send the message as well as the autopilot can)